### DISCUSSION TITLE
Data scaling

### DISCUSSION THREAD

**Discussion by flde** [brianhie/scanorama#124] [Q&A] [answered]
Hi @brianhie, many thanks for the great tool. However, I got a bit lost and hope you can help me out. It is not clear to me at what step I should scale my data. Should I scale the whole data matrix and then split it by batch OR should I split the data matrix by batch and scale each subset?

I tried to understand the source code and think that the data get L2 normalized anyway, right? So I am not sure at all if I should scale data when using the original scanpy implementation? 

I also tried the external scanpy API where they scale all data and perform a PCA on that which is handed to the external scanorama implementation. But the results look strange for whatever reason. 

Highly appreciate your help! Best, Flo

**Comment by brianhie** [comment, marked as answer]
Hi @flde, you could try providing unscaled data given that Scanorama does do an L2 norm. I would encourage you to also provide positive/negative controls into the integration to better determine if integration is working as expected.
